app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -0,0 +1,342 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.list
+
+import android.content.res.Configuration
+import androidx.annotation.DrawableRes
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.defaultMinSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Favicon
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SecondaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+
+val LIST_ITEM_HEIGHT = 56.dp
+
+val ICON_SIZE = 24.dp
+
+/**
+ * List item Composable used to display text with an optional second line of text and
+ * an optional interactable icon at the end.
+ *
+ * @param label The primary text in the item.
```

Comment:
```suggestion
 * @param label The label in the list item.
```
nitpicking a bit on these @param doc since these are our **core** reusable components. I don't think we need to use our internal wording ("primary text") since they are called "Label" in the figma design as well.

Reply:
I agree with this. Also didn't think how calling it "primary text" overlaps with the `PrimaryText' Composable